libs/blocks/merch/merch.js

```diff
@@ -375,7 +384,16 @@ export async function openModal(e, url, offerType) {
   }
 }
 
-export async function getModalAction(offers, options) {
+export function setCtaHash(el, checkoutLinkConfig, offerType) {
+  if (!(el && checkoutLinkConfig && offerType)) return undefined;
+  const authoredHash = checkoutLinkConfig[`${(offerType === OFFER_TYPE_TRIAL) ? FREE_TRIAL_HASH : BUY_NOW_HASH}`];
+  const autogeneratedHash = `${offerType === 'TRIAL' ? 'try' : 'buy'}-${el.closest('merch-card')?.getAttribute('name')}`;
+  const hash = authoredHash ?? autogeneratedHash;
+  el.setAttribute('hash', hash);
```

Comment:
`hash` it not a standard attribute, can you use `data-modal-hash`?

Reply:
I changed to `data-modal-hash` to not mix it with other buttons that use `data-modal-hash`